JSONparser.js

```diff
@@ -0,0 +1,28 @@
+const loggerMessages = require('./loggerMessages');
+
+module.exports = function(rawFile, logger) {
+    
+    // Parsing to [[TRANSACTION-DETAILS], ...]
+    loggerMessages.logDebug('Parsing string as JSON', logger);
+
+    try {
+        const records = JSON.parse(rawFile);
+        const recordsArray = [];
+
+        // Turning JSON ojects into arrays
+        for (let i = 0; i < records.length; i++) {
+            const record = records[i];
+            const currentArray = [record.Date, record.FromAccount, record.ToAccount, record.Narrative, record.Amount];

```

Comment:
Seems like you're going out of your way to make the JSON and XML match the CSV (anonymous arrays). Seems like it'd be nicer to have the CSV match the others i.e. objects with nicely named fields; `{ date, fromAccount, toAccount, narrative, amount }`.